packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_state_var/string_state_var.rs

```diff
@@ -158,3 +158,273 @@ impl StateVarUpdater<String, StringRequiredData> for StringStateVar {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn string_state_var_with_no_dependencies() {
+        // set up data that consists of just the essential variable
+        let mut string_data = StringRequiredData::default();
+        let essential = StateVarMutableView::new_with_value("".to_string(), true);
+        string_data.essential = essential.create_new_read_only_view();
+
+        // add meta data needed for invert to work. Use arbitrary values just to test.
+        string_data._data_query_mapping_data.essential.push((5, 7));
```

Comment:
Why is this set here and not right before the `invert()` call?

Reply:
reworked this